pgmpy/base/DAG.py

```diff
@@ -619,6 +619,88 @@ def get_immoralities(self):
             immoralities[node] = parent_pairs
         return immoralities
 
+    def is_dconnected_moral(self, start, end, observed=None, include_latents=False):
+        """
+        Returns True if `start` and `end` are d-connected given `observed`
```

Comment:
I would suggest a slight change here: `... are d-connected when conditioned on `observed` variables`. I would also remove the moral graph part, as it is a detail related to the algorithm implemented and has nothing to do with the method's functionality.